VS2017 Support

MirzaMerdovic opened the pull request:
Hi aaubry,

I created some changes that are required for this extension to work under VS17 which includes:

- Upgrade  of VS SDK assemblies from version 11 to 15
- Added prerequisite for CoreEditor in manifest file

I tested it under VS17 and VS15 and it appears to be working fine.

Other changes:
- Upgraded YamlDotNet and xUnit to lates versions
- Upgraded .Net to 4.7.1
- Removed some unused project references

Hope this can be useful and thank you for a great extension!

Cheers,
Mirza.

MirzaMerdovic committed:
- Upgrade to .net 4.7.1
- Added prerequisite for CoreEditor in manifest file
- Upgrade of VS SDK assemblies from version 11 to 15

MirzaMerdovic committed:
Upgraded: YamlDotNet and xUnit to latest version.
Removed some unused references.

xt0rted commented on .gitignore:
```diff
@@ -27,3 +27,11 @@ obj/
 [Rr]elease*/
 _ReSharper*/
 [Tt]est[Rr]esult*
+/.vs/YamlDotNet.Editor/v15/Server/sqlite3/storage.ide-wal
```
You should be able to simplify this to just `.vs/`

MirzaMerdovic replied:
Yes, you are right.

MirzaMerdovic committed:
- Trying to fix AppVeyor build failure by using VS SDK from nuget
- gitignore fix

MirzaMerdovic committed:
Trying to fix AppVeyor by removing the Analyzer.

MirzaMerdovic committed:
- Trying to fix AppVeyor build failure by cleaning the Analyzer's residue.

MirzaMerdovic commented:
I am inexperienced when it comes to how AppVeyor works, so I just updated all VS SDK assemblies with nuget packages and removed the Analyzer that was added in the process that was creating an issue during the build.
Unfortunately this now looks probably too messy to be accepted.

aaubry commented:
This looks good. I don't think that using the nuget packages is a problem. Thanks for taking the time to submit this!
I'll test everything tomorrow and if there are no issues, I'll release the updated editor package.

MirzaMerdovic commented:
Cool, please let me know if I can help if any issues arise or otherwise.

aaubry commented:
I am unable to install .NET 4.7.1 on my machine. Is this upgrade really necessary ?

MirzaMerdovic commented:
No, and the lowest version that built successfully for me is 4.6.1

aaubry commented:
Hi, sorry about the late response.
So, I changed to .NET 4.6.1, and was able to compile the package. I am able to install it on VS2017, but not on VS2015. Do you have an idea on what is wrong?

MirzaMerdovic commented:
Hi, no problem.
No, I will see with couple of my friends that tested this for me on VS15 and see if they have problems with this build on .net 4.6.1, since I don't have VS15 on my laptop anymore, but in worst case I will install it.
I will get back to you during the next week if that is ok.

MirzaMerdovic commented:
Hi, sorry about no update I was quite busy last week and than holidays arrived, I will try to look into this during this week.

MirzaMerdovic commented:
Hey, sorry for not doing anything about this for a long time, but currently I can't get enough time to investigate the issues you mention, so I will close the PR and hopefully get back to it in the near future.
